### PR TITLE
BUG: Fix IntervalIndex.to_tuples() with NA values

### LINKED ISSUE #18756: BUG: IntervalIndex.to_tuples doesn't handle NA correctly
#### Code Sample, a copy-pastable example if possible

```python
In [2]: idx = pd.interval_range(0, 3).insert(0, np.nan)
   ...: idx
   ...:
Out[2]:
IntervalIndex([nan, (0.0, 1.0], (1.0, 2.0], (2.0, 3.0]]
              closed='right',
              dtype='interval[float64]')

In [3]: idx.to_tuples()
Out[3]: Index([(nan, nan), (0.0, 1.0), (1.0, 2.0), (2.0, 3.0)], dtype='object')
```
#### Problem description
A tuple of `np.nan` is returned instead of just `np.nan`.  Note that `values` and `tolist` both yield `np.nan`:

```python
In [4]: idx.values
Out[4]:
array([nan, Interval(0.0, 1.0, closed='right'),
       Interval(1.0, 2.0, closed='right'),
       Interval(2.0, 3.0, closed='right')], dtype=object)

In [5]: idx.tolist()
Out[5]:
[nan,
 Interval(0.0, 1.0, closed='right'),
 Interval(1.0, 2.0, closed='right'),
 Interval(2.0, 3.0, closed='right')]
```

#### Expected Output
`np.nan` instead of a tuple of `np.nan`:
```python
Index([nan, (0.0, 1.0), (1.0, 2.0), (2.0, 3.0)], dtype='object')
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: d2fd22e1f995a638f1e84b0ef091eaf70e171fdc
python: 3.6.1.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.22.0.dev0+353.gd2fd22e
pytest: 3.1.2
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.13.1
scipy: 0.19.1
pyarrow: 0.6.0
xarray: 0.9.6
IPython: 6.1.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: 3.4.2
numexpr: 2.6.2
feather: 0.4.0
matplotlib: 2.0.2
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 0.9.8
lxml: 3.8.0
bs4: None
html5lib: 0.999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: 0.1.0
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by jschendel**
- [X] closes #18756
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/core/indexes/interval.py [hunk 1]
```diff
@@ -546,7 +546,11 @@ def from_tuples(cls, data, closed='right', name=None, copy=False):
 
     def to_tuples(self):
         """Return an Index of tuples of the form (left, right)"""
```
Comment (jreback) [inline pandas/core/indexes/interval.py]: I guess would be ok to add a kwarg ``na_tuple=False`` here to select behavior as @shoyer suggests.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jschendel** [comment]
Added the `na_tuple` kwarg.  Had to do some explicit checks of the NA value in the test, due equality issues with tuples of NA (see https://github.com/pandas-dev/pandas/issues/18756#issuecomment-351881631 for an example)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jschendel

**Comment by shoyer** [comment]
Indeed, thanks!

**Pull request merged**